### DISCUSSION TITLE
jotai import bug

### DISCUSSION THREAD

**Discussion by 01-binary** [pmndrs/jotai#2459] [Bug report]
Hi I'm using jotai in Nextjs 13, VScode, Jotai ^2.7.1
When I import jotai lib, i have trouble like below picture.
in VScode, import auto complete doesn't work.
i can't see import from 'jotai'
i wonder what happen..
if you guys know the answer please comment!
thank you 
![image](https://github.com/pmndrs/jotai/assets/15652602/15feb8fb-3c06-4f1e-ad76-34f135071fa8)

**Comment by dai-shi** [comment]
Thanks for reporting. I'm not sure how it can be helped from our side. Hope someone with expertise with vscode could help.

**Reply by R-Bower** [reply]
I am seeing something similar from WebStorm.  The auto-import functionality does not suggest the correct import path.

This import would cause an error:
![image](https://github.com/pmndrs/jotai/assets/11933208/b2e26b22-ecc9-4f97-9e24-9795f7824dc1)

The correct import is `import {Provider} from "jotai"`

In my VSCode, the correct import is automatically detected (along with a bunch of incorrect ones):
![image](https://github.com/pmndrs/jotai/assets/11933208/76f750ed-a067-4b89-a033-79ba09efcdb9)

In my experience, this typically indicates an issue with the package.json `types` entry.  I see that jotai is also using [exports](https://github.com/pmndrs/jotai/blob/main/package.json#L34) and `typeVersions`.  I'm a bit stumped here myself.

Does this warrant a bug report?

**Reply by R-Bower** [reply]
If I delete the `typesVersions` entry of package.json, the import shows up as expected in WebStorm:
![image](https://github.com/pmndrs/jotai/assets/11933208/249cc34c-36f7-4573-bfac-c39d48a78a87)

Not ideal, but an interesting find.  I'll do some more tinkering.

**Reply by 01-binary** [reply]
oh.. interesting! thank you for letting me know

**Reply by 01-binary** [reply]
The project I'm having trouble with uses yarn berry.
but in other project using yarn 1 or npm, that project have no issue.

**Comment by dai-shi** [comment]
@R-Bower 

> Does this warrant a bug report?

This #2459 is the bug report. So, it's fine.

> If I delete the `typesVersions` entry of package.json

Nice finding! But I think we need to keep `typesVersions`. Does the order in package.json matter?
